### PR TITLE
do not fetch images with empty list

### PR DESCRIPTION
**Pull request by vetinari**
empty list means all images -> full CPU for ages...

### CODE REVIEW

**Comment by vetinari** [comment]
:+1:

File: tests/test_aws.py [hunk 1]
```diff
@@ -370,6 +370,26 @@ def test_aws_get_running_apps(monkeypatch):
     boto.assert_called_with('ec2', region_name=REGION)
 
 
+def test_aws_get_running_apps_existing(monkeypatch):
```
Comment (mohabusama) [inline tests/test_aws.py]: You are not making the actual call here to ``aws.get_running_apps`` and asserting the result 😉

File: zmon_aws_agent/aws.py [hunk 2]
```diff
@@ -300,19 +302,21 @@ def get_running_apps(region, existing_entities=None):
...
-                for img in imgs:
-                    if img['ImageId'] == i['image']['id']:
-                        i['image']['name'] = img.get('Name', 'UNKNOWN')
-                        date = img.get('CreationDate', '1970-01-01T00:00:00.000+00:00').replace('Z', '+00:00')
-                        i['image']['date'] = date
-                        break
-        except:
-            logger.exception('Failed to retrieve image descriptions')
+        # prevent fetching all images (in case the images is empty, it will do so):
+        if len(list(images)) > 0:
```
Comment (mohabusama) [inline zmon_aws_agent/aws.py]: ``if images:`` should work

**Comment by vetinari** [comment]
:+1:

**Comment by vetinari** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**